dev/com.ibm.ws.microprofile.reactive.streams.operators/src/com/ibm/ws/microprofile/reactive/streams/WASReactiveStreamsEngineImpl.java

```diff
@@ -36,11 +36,20 @@
 
     private static ReactiveStreamsEngine singleton = null;
 
-    @Reference(target = "(component.name=com.ibm.ws.threading)")
-    private static ExecutorService serverExecutor;
+    private static ExecutorService serverExecutor = null;
 
     private static ExecutorService executor = null;
 
+    /**
+     * Declarative Services method for setting the Liberty executor.
+     *
+     * @param svc the service
+     */
+    @Reference(target = "(component.name=com.ibm.ws.threading)")
+    protected void setExecutor(ExecutorService svc) {
+        serverExecutor = svc;
+    }
```

Comment:
This does work, but it may have the unintended side effect that the instance you return from `getEngine()` will be a different instance from the one you get if you look up the OSGi component.

As this class doesn't really have any state, I don't think this will actually cause you any problems, but I'm not sure it's quite what you intended.

Reply:
I will raise and issue to have an @Activate method that will assign "this" into the singleton:
https://github.com/OpenLiberty/open-liberty/issues/4497